plugins/proxy-backend/src/service/router.ts

```diff
@@ -186,8 +186,34 @@ export async function createRouter(
   const { pathname: pathPrefix } = new URL(externalUrl);
 
   const proxyConfig = options.config.getOptional('proxy') ?? {};
+  configureMiddlewares(options, router, pathPrefix, proxyConfig);
 
-  Object.entries(proxyConfig).forEach(([route, proxyRouteConfig]) => {
+  if (options.config.subscribe) {
+    let currentKey = JSON.stringify(proxyConfig);
+
+    options.config.subscribe(() => {
+      const newProxyConfig = options.config.getOptional('proxy') ?? {};
+      const newKey = JSON.stringify(newProxyConfig);
+
+      if (currentKey !== newKey) {
+        currentKey = newKey;
+
+        router.stack = [];
```

Comment:
This is as far as I could find the only way to clear it up. The `use()` calls adds handlers to this `stack` array without any side effects: https://github.com/expressjs/express/blob/4.x/lib/router/index.js#L478

It sure isn't pretty, but I didn't find any other way. A more clean approach would maybe be to recreate the router, but the reference to the rest of the backend wouldn't get updated then as far as I understand.

Reply:
So I finally managed to figure out how to test the new implementation properly. It turned out better all together with tests now testing actual requests against the proxy instead of attempting to validate configured routes through implementation details.

PTAL @Rugvip